bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java

```diff
@@ -0,0 +1,167 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * Handler for security cameras.
+ * <p>
+ * This implementation handles services and commands that are common to all cameras, which are currently:
+ * 
+ * <ul>
+ * <li><code>PrivacyMode</code> - Controls whether the camera records images</li>
+ * <li><code>CameraNotification</code> - Enables or disables notifications for the camera</li>
+ * </ul>
+ * 
+ * <p>
+ * The Eyes outdoor camera advertises a <code>CameraLight</code> service, which unfortunately does not work properly.
+ * Valid states are <code>ON</code> and <code>OFF</code>.
+ * One of my two cameras returns <code>HTTP 204 (No Content)</code> when requesting the state.
+ * Once Bosch supports this service properly, a new subclass may be introduced for the Eyes outdoor camera.
+ * 
+ * @author David Pace - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class CameraHandler extends BoschSHCHandler {
+
+    private PrivacyModeService privacyModeService;
+    private CameraNotificationService cameraNotificationService;
+
+    public CameraHandler(Thing thing) {
+        super(thing);
+        this.privacyModeService = new PrivacyModeService();
+        this.cameraNotificationService = new CameraNotificationService();
+    }
+
+    @Override
+    protected void initializeServices() throws BoschSHCException {
+        super.initializeServices();
+
+        this.registerService(this.privacyModeService, this::updateChannels, List.of(CHANNEL_PRIVACY_MODE));
+        this.registerService(this.cameraNotificationService, this::updateChannels,
+                List.of(CHANNEL_CAMERA_NOTIFICATION));
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        requestInitialStates();
+    }
+
+    /**
+     * Requests the initial states for relevant services.
+     * <p>
+     * If this is not done, items associated with the corresponding channels with stay in an uninitialized state
+     * (<code>null</code>).
+     * This in turn leads to events not being fired properly when switches are used in the UI.
+     * <p>
+     * Unfortunately the long poll results do not contain camera-related updates, so this is the current approach
+     * to get the initial states.
+     */
+    private void requestInitialStates() {
+        requestInitialPrivacyState();
+        requestInitialNotificationState();
+    }
+
+    private void requestInitialPrivacyState() {
+        try {
+            @Nullable
+            PrivacyModeServiceState serviceState = privacyModeService.getState();
+            if (serviceState != null) {
+                super.updateState(CHANNEL_PRIVACY_MODE, serviceState.value.toOnOffType());
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.error("Error while requesting initial privacy state of camera {}", getBoschID(), e);
```

Comment:
Thanks for calling my attention to the logging guidelines. In this case this would indeed not be a bug and the log level `error` is not applicable here. I changed the log level for all logger calls in those methods to `debug` now. I am not sure whether a stack trace would be allowed or even useful on debug level. In my current commit I have omitted the stack trace. If you are fine with that you can resolve this comment.

Reply:
It's really fine without stack trace, especially for debug level. Nice catch about error level, @GerdZanker, I somehow missed that.